File: app/src/main/java/br/com/algorit/mangaalert/ui/recyclerview/MangaRecyclerAdapter.java
package br.com.algorit.mangaalert.ui.recyclerview;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import br.com.algorit.mangaalert.R;
import br.com.algorit.mangaalert.roomdatabase.model.Manga;

public class MangaRecyclerAdapter extends RecyclerView.Adapter<MangaRecyclerAdapter.MyViewHolder> {

    private final ItemClickListener clickListener;
    private final LayoutInflater layoutInflater;
    private List<Manga> listaManga;

    public MangaRecyclerAdapter(ItemClickListener clickListener, Context context) {
        this.clickListener = clickListener;
        layoutInflater = LayoutInflater.from(context);
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = layoutInflater.inflate(R.layout.item_lista, parent, false);
        return new MyViewHolder(view, clickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        if (listaManga != null) {
            Manga manga = listaManga.get(position);
            holder.titulo.setText(manga.getNome());
            holder.checkBox.setOnCheckedChangeListener(((buttonView, isChecked) ->
                    manga.setChecked(isChecked)));
            holder.checkBox.setChecked(manga.isChecked());
        }
    }

    public void setMangas(List<Manga> mangas) {
        this.listaManga = mangas;
        if (listaManga != null) {
            Collections.sort(this.listaManga, Manga::compareTo);
        }
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        if (this.listaManga != null)
            return listaManga.size();
        else return 0;
    }

    public List<Manga> getCheckedItems() {
        List<Manga> checkedItems = new ArrayList<>();
        for (Manga manga : listaManga) {
            if (manga.isChecked())
                checkedItems.add(manga);
        }
        return checkedItems;
    }

    static class MyViewHolder extends RecyclerView.ViewHolder {

        private CheckBox checkBox;
        private TextView titulo;
        private ItemClickListener clickListener;

        MyViewHolder(View itemView, ItemClickListener clickListener) {
            super(itemView);

            checkBox = itemView.findViewById(R.id.item_lista_check);
            titulo = itemView.findViewById(R.id.item_lista_titulo);

//            itemView.setOnClickListener(view -> clickListener.onItemClick(view, getAdapterPosition()));
        }
    }
}

File: app/src/main/java/br/com/algorit/mangaalert/ui/recyclerview/MangaRecyclerAdapter.kt
package br.com.algorit.mangaalert.ui.recyclerview

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.CompoundButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import br.com.algorit.mangaalert.R
import br.com.algorit.mangaalert.roomdatabase.model.Manga
import java.util.*

class MangaRecyclerAdapter(context: Context?) :
    RecyclerView.Adapter<MangaRecyclerAdapter.MyViewHolder>() {
    private val layoutInflater: LayoutInflater = LayoutInflater.from(context)
    private var listaManga: List<Manga>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = layoutInflater.inflate(R.layout.item_lista, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        if (listaManga != null) {
            val manga = listaManga!![position]
            holder.titulo.text = manga.nome
            holder.checkBox.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
                manga.isChecked = isChecked
            }
            holder.checkBox.isChecked = manga.isChecked
        }
    }

    fun setMangas(mangas: List<Manga>) {
        listaManga = mangas
        if (listaManga != null) {
            Collections.sort(listaManga) { obj: Manga, manga: Manga -> obj.compareTo(manga) }
        }
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return if (listaManga != null) listaManga!!.size else 0
    }

    val checkedItems: List<Manga>
        get() {
            val checkedItems: MutableList<Manga> = ArrayList()
            for (manga in listaManga!!) {
                if (manga.isChecked) checkedItems.add(manga)
            }
            return checkedItems
        }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val checkBox: CheckBox = itemView.findViewById(R.id.item_lista_check)
        val titulo: TextView = itemView.findViewById(R.id.item_lista_titulo)
    }

}